hail/src/main/scala/is/hail/expr/ir/InferPType.scala

```diff
@@ -190,55 +190,56 @@ object InferPType {
         PDict(elt.types(0), PArray(elt.types(1)), collection.pType2.required)
       case ArrayMap(a, name, body) =>
         infer(a)
-        infer(body, env.bind(name, a.pType2.asInstanceOf[PArray].elementType))
+        infer(body, env.bind(name, a.pType2.asInstanceOf[PStreamable].elementType))
```

Comment:
all of these PStreamable casts can just be PStream. Would be a good check, actually.